### PR TITLE
RIA-4116: Added condition to ignore sending Home Office notifications for inflight cases and for cases where validation has failed

### PR DESCRIPTION
**Pull request by jkumar-moj**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RIA-4116

### Change description ###
RIA-4116: Added condition to ignore sending Home Office notifications for inflight cases and for cases where validation has failed

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by makowalski — APPROVED** [review]
LGTM

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/HomeOfficeCaseNotificationsHandler.java [hunk 1]
```diff
@@ -86,11 +88,33 @@ public boolean canHandle(
...
+        final YesOrNo homeOfficeNotificationsEligible = asylumCaseWithHomeOfficeData.read(HOME_OFFICE_NOTIFICATIONS_ELIGIBLE, YesOrNo.class)
+            .orElse(YesOrNo.NO);
 
         if (asylumCaseWithHomeOfficeData.read(APPELLANT_IN_UK, YesOrNo.class).map(
             value -> value.equals(YesOrNo.YES)).orElse(true)) {
 
-            asylumCaseWithHomeOfficeData = homeOfficeApi.call(callback);
+            if ("SUCCESS".equalsIgnoreCase(homeOfficeSearchStatus)
+                && homeOfficeNotificationsEligible == YesOrNo.YES) {
+
```
Comment (makowalski) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/HomeOfficeCaseNotificationsHandler.java:102]: Please add info log with caseId, Home Office number, that we are about to send HO notification

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/HomeOfficeCaseNotificationsHandler.java [hunk 2]
```diff
@@ -86,11 +88,33 @@ public boolean canHandle(
...
+                asylumCaseWithHomeOfficeData = homeOfficeApi.call(callback);
+
```
Comment (makowalski) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/HomeOfficeCaseNotificationsHandler.java:104]: Please add info log with caseId, Home Office number, that we sent HO notification successfully

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/HomeOfficeCaseNotificationsHandler.java (hunk 1 above)
Comment (jkumar-moj) [reply src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/HomeOfficeCaseNotificationsHandler.java:111]: log entry added

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/HomeOfficeCaseNotificationsHandler.java (hunk 2 above)
Comment (jkumar-moj) [reply src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/HomeOfficeCaseNotificationsHandler.java:117]: log entry added

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/HomeOfficeCaseNotificationsHandler.java [hunk 3]
```diff
@@ -86,11 +88,33 @@ public boolean canHandle(
...
                 .getCaseData();
 
+        final String homeOfficeSearchStatus = asylumCaseWithHomeOfficeData.read(HOME_OFFICE_SEARCH_STATUS, String.class)
+            .orElse("");
+
+        final YesOrNo homeOfficeNotificationsEligible = asylumCaseWithHomeOfficeData.read(HOME_OFFICE_NOTIFICATIONS_ELIGIBLE, YesOrNo.class)
+            .orElse(YesOrNo.NO);
 
         if (asylumCaseWithHomeOfficeData.read(APPELLANT_IN_UK, YesOrNo.class).map(
             value -> value.equals(YesOrNo.YES)).orElse(true)) {
```
[the remark this answers is not in the document]
Comment (jkumar-moj) [reply src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/HomeOfficeCaseNotificationsHandler.java:107]: log entry added

**Review by makowalski — APPROVED** [review]
LGTM

**Review by Srihari-Nakkala — APPROVED** [review]
LGTM

**Review by makowalski — APPROVED** [review]
LGTM

**Pull request merged**